### ISSUE TITLE
ecs plugin: configuring listener_arn, listener port, protocol and cert are reset 

### ISSUE THREAD

**Issue by lukemelia** [hashicorp/waypoint#1568]
**Describe the bug**
When using the ecs plugin with a deploy config similar to:

```
    deploy {
        use "aws-ecs" {
            ...
            alb {
                listener_arn = "arn:aws:elasticloadbalancing:us-east-1:680542703984:listener/app/hub-staging/etc/etc"
            }
        }
    }
```

And the referenced listener has a protocol set to HTTPS and port set to 443, the plugin changes the protocol to HTTP and the port to 80 and removes the certificate reference.

**Expected behavior**
No settings of the existing listener except the default actions should be changed.

**Waypoint Platform Versions**
* Waypoint CLI Version: 0.3.0
* Waypoint Server Platform and Version: docker
* Waypoint Plugin: aws/ecs

**Additional context**

The update appears to be happening here: https://github.com/hashicorp/waypoint/blob/dca533f0e2a36bbc8df6fa89fbfa475bb7f673e3/builtin/aws/ecs/platform.go#L811-L813

It is only possible to have the `port` and `protocol` values referenced in that code to be correct if you specify a certificate_id (https://github.com/hashicorp/waypoint/blob/dca533f0e2a36bbc8df6fa89fbfa475bb7f673e3/builtin/aws/ecs/platform.go#L670-L682)

However, it is not allowed to specify a certificate_id and a listener_arn (https://github.com/hashicorp/waypoint/blob/dca533f0e2a36bbc8df6fa89fbfa475bb7f673e3/builtin/aws/ecs/platform.go#L50-L52)

**Comment by evanphx** [comment]
Yup! The code is incorrectly modifying the listener in this case. The code here: https://github.com/hashicorp/waypoint/blob/dca533f0e2a36bbc8df6fa89fbfa475bb7f673e3/builtin/aws/ecs/platform.go#L795-L828 should be moved up to line 761, ie within the scope of seeing a listener than a deploy previously created.

**Comment by lukemelia** [comment]
We have sorta patched this with an after hook, but it still results in about 1 second of downtime when we deploy. Any chance of getting this addressed soon?

**Comment by catsby** [comment]
Hey @lukemelia - thank you for your patience here. I circled back to look at this but unfortunately the code has changed considerably since this was opened and @evanphx gave his recommendation. 

I imagine there's still an issue here of course. Could you verify for us what version of Waypoint you're using?

**Comment by lukemelia** [comment]
Thanks for getting back to me on this @catsby. We're still on 0.4 and had been planning to upgrade concurrent with this fix. I'll upgrade today though and confirm whether we're still experiencing this on 0.7. Stay tuned.

**Comment by catsby** [comment]
Thanks so much @lukemelia. Reading through the code I wont be surprised if it is still there. I'm trying to get some cert things settled on our end so I can replicate. Let us know!

**Comment by lukemelia** [comment]
@catsby I upgraded to 0.7.0 and ran the deploy. The bug described here is still present. Here is my ALB listener in the AWS console before I attempt to deploy:

![image](https://user-images.githubusercontent.com/353/150447560-53d9a24b-c07d-4742-8a65-14f70808ecf1.png)

And here it is after the deployment stage:

![image](https://user-images.githubusercontent.com/353/150447601-710f247b-3955-41bb-9e51-7316a7274784.png)

In addition (and I don't know if this is related to specifying the alb.listener_arn property), the release stage is failing with the following error:

```
» Releasing hub...
! ValidationError: A load balancer ARN must be specified
        status code: 400, request id: 4b3e80a1-0f67-4eb5-9315-790dda9ffe16
```
